src/test/java/uk/gov/hmcts/reform/civil/service/stitching/CivilDocumentStitchingServiceTest.java

```diff
@@ -0,0 +1,128 @@
+                              .build())
+            .build();
+    //Optional should be removed after refactoring
+    private final Optional<String> stitchStatus = Optional.of(randomAlphabetic(8));
+    private final Optional<Document> stitchedDocument = Optional.of(mock(Document.class));
+
+    private List<DocumentMetaData> buildDocumentMetaDataList() {
+        return List.of(
+            new DocumentMetaData(
+                CLAIM_FORM.getDocumentLink(),
+                "Sealed Claim Form",
+                LocalDate.now().toString()
+            )
+        );
+    }
+
+    public CaseData buildCaseData(Bundle bundle){
+        List<IdValue<Bundle>> caseBundles = List.of(new IdValue<>("1", bundle));
+        return CaseDataBuilder.builder().caseBundles(caseBundles).build();
+    }
+    @Test
+    public void whenCaseDataIsNullWithPayloadAndNoCaseBundlesNoStitchingDocument() {
+        List<DocumentMetaData> documentMetaDataList = new ArrayList<>();
+        CaseData caseData = CaseDataBuilder.builder().legacyCaseReference("ClaimNumber").build();
+        CaseDocument caseDocument = civilDocumentStitchingService.bundle(documentMetaDataList, "Auth",
+                                                                  "Title", "FileName", caseData);
+        Assertions.assertNull(caseDocument);
+    }
+    @Test
+    public void whenCaseDataIsNullWithNoCaseBundles() {
+        List<DocumentMetaData> documentMetaDataList = buildDocumentMetaDataList();
+        CaseData caseData = CaseDataBuilder.builder().legacyCaseReference("ClaimNumber").build();
+        when(stitchingConfiguration.getStitchingUrl()).thenReturn("dummy_url");
+        CaseDocument caseDocument = civilDocumentStitchingService.bundle(documentMetaDataList, "Auth",
+                                                                         "Title", "FileName", caseData);
+        Assertions.assertNull(caseDocument);
+    }
+    @Test
+    public void whenCaseDataIsNotNullWithCaseBundlesAndNoStitchingDocumentThenCaseDocumentIsNull() {
+        List<DocumentMetaData> documentMetaDataList = buildDocumentMetaDataList();
+        CaseData caseData = CaseDataBuilder.builder().legacyCaseReference("ClaimNumber").build();
+        CaseData caseData1 = buildCaseData(mock(Bundle.class));
+        when(stitchingConfiguration.getStitchingUrl()).thenReturn("dummy_url");
+        when(bundleRequestExecutor.post(any(BundleRequest.class), anyString(), anyString())).thenReturn(caseData1);
+        CaseDocument caseDocument = civilDocumentStitchingService.bundle(documentMetaDataList, "Auth",
+                                                                         "Title", "FileName", caseData);
+        Assertions.assertNull(caseDocument);
+    }
+    @Test
+    public void whenStitchingDocumentIsPresentAndCaseDataNotNullThenCaseDocumentNotNull() {
+        List<DocumentMetaData> documentMetaDataList = buildDocumentMetaDataList();
+        Bundle bundle = new Bundle(
```

Comment:
For consistency, use the builder here. It helps clarifying what fields you are setting.

Reply:
Done